From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sun, 23 Jul 2023 18:25:11 +0200
Subject: [PATCH] Client perspective - Localization - Register server-side
 translations

License: Fiddle Public License Agreement 1.0 (included in license/FPL-1.0.txt)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 0a46669503131fef418a4cd200061bc4d9c50495..f5a2c2dede0ce27db7e514f7c20264fa7256f4ce 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -73,6 +73,7 @@ import net.minecraft.world.level.storage.LevelSummary;
 import net.minecraft.world.level.storage.PrimaryLevelData;
 import net.minecraft.world.level.storage.WorldData;
 import org.bukkit.Material;
+import org.fiddlemc.fiddle.packet.locale.ServerSideTranslations;
 import org.fiddlemc.fiddle.server.FiddleFeatureSupport;
 import org.fiddlemc.fiddle.configuration.FiddleConfigurations;
 import org.fiddlemc.fiddle.configuration.PermanenceConfirmation;
@@ -236,6 +237,18 @@ public class Main {
             Path path1 = Paths.get("server.properties");
             DedicatedServerSettings dedicatedserversettings = new DedicatedServerSettings(optionset); // CraftBukkit - CLI argument support
 
+            // Fiddle start - client perspective - localization - register server-side translations
+            // TODO this is not read from the resource packs yet
+            for (var block : BuiltInRegistries.BLOCK) {
+                if (block.isAddedByFiddle()) {
+                    ServerSideTranslations.registerAddedTranslation("en_us", block.getDescriptionId(), "Fiddle " + block.getKeyInBlockRegistry().getPath());
+                    ServerSideTranslations.registerAddedTranslation("en_gb", block.getDescriptionId(), "Fiddle " + block.getKeyInBlockRegistry().getPath() + ", I say");
+                    ServerSideTranslations.registerAddedTranslation("zh_cn", block.getDescriptionId(), "小提琴 " + block.getKeyInBlockRegistry().getPath());
+                }
+            }
+            ServerSideTranslations.registerVanillaTranslationReplacement("en_us", "block.minecraft.bookshelf", "Oak Bookshelf");
+            // Fiddle end - client perspective - localization - register server-side translations
+
             // Fiddle start - modifiable Bukkit enums - inject runtime versions - common utilities
             ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
             class InjectRuntimeClassVersionHelper {
